adala/runtimes/base.py

```diff
@@ -81,11 +88,27 @@ def batch_to_batch(
             field_schema (Optional[Dict]): Field JSON schema to use in the templates. Defaults to all fields are strings,
                 i.e. analogous to {"field_n": {"type": "string"}}.
             instructions_first (bool): Whether to put instructions first. Defaults to True.
-
+            run_parallel (bool): Whether to run batch processing in parallel. Defaults to True.
         Returns:
             InternalDataFrame: The processed batch.
         """
-        output = batch.progress_apply(
+        if self.concurrency == -1:
+            # run batch processing each row in a parallel way, using all available CPUs
+            logger.info("Running batch processing in parallel using all available CPUs")
+            apply_func = batch.parallel_apply
+        elif self.concurrency == 1:
+            # run batch processing each row in a sequential way
+            logger.info("Running batch processing sequentially")
+            apply_func = batch.progress_apply
+        elif self.concurrency > 1:
+            # run batch processing each row in a parallel way, using a fixed number of CPUs
+            logger.warning(f'Setting exact number of CPUs for parallel processing is not supported yet. '
```

Comment:
seems that the number of workers pandarallel uses is governed by the `nb_workers` param for `pandarallel.initialize`. Can we just wait until this point to run `pandarallel.initialize` (i.e. once we know the desired number of workers), or does that break for some reason?

Reply:
I thought about this, but running it every time will introduce an overhead on every call. But maybe you're right, let's make it for now, then we will optimize it in a future